pandas/_typing.py

```diff
@@ -125,12 +125,13 @@
 ]
 
 # dtypes
-NpDtype = Union[str, np.dtype, type_t[Union[str, complex, bool, object]]]
+# using type: ignore due to incompatible types in assignment
+NpDtype = Union[str, "np.dtype[np.generic]", type_t[Union[str, complex, bool, object]]]  # type: ignore[assignment]
 Dtype = Union["ExtensionDtype", NpDtype]
 AstypeArg = Union["ExtensionDtype", "npt.DTypeLike"]
 # DtypeArg specifies all allowable dtypes in a functions its dtype argument
 DtypeArg = Union[Dtype, Dict[Hashable, Dtype]]
-DtypeObj = Union[np.dtype, "ExtensionDtype"]
+DtypeObj = Union["np.dtype[np.generic]", "ExtensionDtype"]  # type: ignore[assignment]
```

Comment:
You don't need these two ignores, see the mypy output:

>  pandas/_typing.py:129: error: Unused "type: ignore" comment
> pandas/_typing.py:134: error: Unused "type: ignore" comment

but you need an ignore here:


> pandas/core/dtypes/cast.py:222: error: Incompatible types in assignment (expression has type "Union[generic, Any]", variable has type "Union[Union[str, float, bool], Union[Period, Timestamp, Timedelta, Interval[Any]], datetime64, timedelta64, datetime]")  [assignment]

After removing/adding those ignores, you can run `pre-commut run -a` to make sure that black&flake8&... are happy :) (or just run `pre-commut run black -a` and `pre-commut run flake8 -a`)